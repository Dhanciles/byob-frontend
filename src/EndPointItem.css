pre {
  padding: 10px; 
}

h3 {
  font-size: 2rem; 
}

p {
  font-size: 1.5rem;
}

.example-text {
  font-size: 1.5rem; 
  font-weight: 600; 
}