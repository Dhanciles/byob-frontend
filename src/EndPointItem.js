import React, { Component } from 'react';  

export default class EndPointItem extends Component {
  constructor(props) {
  super(props)
  this.state = {
    selected: false
    }
  }

  handleClick = (event) => {
    this.setState({
      selected: !this.state.selected
    })
  }
  
  render() {
    const { method, path, examples, successful_status_code, returns } = this.props.data;
    let itemBody = '';
    let selectedClass = '';
    if(this.state.selected) {
      itemBody = (
        <div className="info-area">
          <h3>Example {Object.keys(examples)[0]}:</h3>
          <pre>{examples.request || 'None'}</pre>
          <p>Successful Status Code: <span>{successful_status_code}</span></p>
          <p>This endpoint returns: {returns}</p>
          <h3>Example {Object.keys(examples)[1]}:</h3>
          <pre>{examples.response}</pre>
        </div>
      )
      selectedClass = 'selected';
    }

    return (
      <div className={`endpoint-item-container ${selectedClass}`}>
        <div className="button-and-path-container">
          <button onClick={this.handleClick}>{method}</button>
          <span className="endpoint-text-container">
            <p className="url-path">{path}</p>
          </span>
        </div>
        {itemBody}
      </div>
    );
  }
}