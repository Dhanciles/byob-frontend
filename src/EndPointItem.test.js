import React from 'react'; 
import { shallow } from 'enzyme'; 
import EndPointItem from './EndPointItem.js'
import endPointsData from '../endpoints-data.js'


describe('EndPointItem', () => {
  let mockData
  beforeEach(() => {
    mockData = endPointsData.endpoints
  })
  it('should match snapshot', () => {
    const wrapper = shallow(<EndPointItem data={ mockData }/>)

    expect(wrapper).toMatchSnapshot()
  })
  
})