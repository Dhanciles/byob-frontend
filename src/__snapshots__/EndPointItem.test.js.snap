// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EndPointItem should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EndPointItem
    data={
      Array [
        Object {
          "examples": Object {
            "response": Array [
              Object {
                "capital": false,
                "created_at": "2019-02-05T00:54:20.896Z",
                "id": 1,
                "name": "New York City",
                "population": 8622698,
                "state": "NY",
                "updated_at": "2019-02-05T00:54:20.896Z",
              },
              Object {
                "capital": false,
                "created_at": "2019-02-05T00:54:20.906Z",
                "id": 3,
                "name": "Los Angeles",
                "population": 3999759,
                "state": "CA",
                "updated_at": "2019-02-05T00:54:20.906Z",
              },
              Object {
                "capital": false,
                "created_at": "2019-02-05T00:54:20.906Z",
                "id": 2,
                "name": "Chicago",
                "population": 2716450,
                "state": "IL",
                "updated_at": "2019-02-05T00:54:20.906Z",
              },
            ],
          },
          "method": "GET",
          "path": "/api/v1/cities",
          "properties": Object {
            "capital": "boolean",
            "created_at": "timestamp",
            "id": "number",
            "name": "string",
            "population": "number",
            "state": "string",
            "updated_at": "timestamp",
          },
          "request_should_include": null,
          "returns": "an array of all cities in the database",
          "successful_status_code": 200,
        },
        Object {
          "examples": Object {
            "request": Object {
              "capital": false,
              "name": "New York City",
              "population": 8622698,
              "state": "NY",
            },
          },
          "method": "POST",
          "path": "/api/v1/cities",
          "properties": Object {
            "capital": "boolean",
            "created_at": "timestamp",
            "id": "number",
            "name": "string",
            "population": "number",
            "state": "string",
            "updated_at": "timestamp",
          },
          "request_should_include": Array [
            "name",
            "state",
            "population",
            "capital",
          ],
          "returns": "all properties with new id",
          "successful_status_code": 201,
        },
        Object {
          "examples": Object {
            "response": Array [
              Object {
                "address": "Chelsea Market, 75 9th Avenue, New York 10011",
                "avg_cost": "$25",
                "city": "New York City",
                "city_id": 1,
                "created_at": "2019-02-05T00:54:20.932Z",
                "id": 1,
                "name": "Los Tacos No. 1",
                "rating": 4,
                "updated_at": "2019-02-05T00:54:20.932Z",
              },
              Object {
                "address": "1556 2Nd Avenue 10028",
                "avg_cost": "$60",
                "city": "New York City",
                "city_id": 1,
                "created_at": "2019-02-05T00:54:20.935Z",
                "id": 3,
                "name": "Cascabel Taqueria",
                "rating": 4,
                "updated_at": "2019-02-05T00:54:20.935Z",
              },
              Object {
                "address": "303 West 48th Street, New York 10036",
                "avg_cost": "$65",
                "city": "New York City",
                "city_id": 1,
                "created_at": "2019-02-05T00:54:20.936Z",
                "id": 2,
                "name": "Sombrero",
                "rating": 2,
                "updated_at": "2019-02-05T00:54:20.936Z",
              },
            ],
          },
          "method": "GET",
          "path": "/api/v1/restaurants",
          "properties": Object {
            "address": "string",
            "avg_cost": "string",
            "city_id": "number",
            "created_at": "timestamp",
            "id": "number",
            "name": "string",
            "rating": "number",
            "updated_at": "timestamp",
          },
          "request_should_include": null,
          "returns": "an array of all restaurants in the database",
          "successful_status_code": 200,
        },
        Object {
          "examples": Object {
            "response": Array [
              Object {
                "capital": true,
                "created_at": "2019-02-05T00:54:20.920Z",
                "id": 19,
                "name": "Denver",
                "population": 704621,
                "state": "CO",
                "updated_at": "2019-02-05T00:54:20.920Z",
              },
            ],
          },
          "method": "GET",
          "path": "/api/v1/cities/:id",
          "properties": Object {
            "capital": "boolean",
            "created_at": "timestamp",
            "id": "number",
            "name": "string",
            "population": "number",
            "state": "string",
            "updated_at": "timestamp",
          },
          "request_should_include": null,
          "returns": "an array of any cities in the database that match the given id",
          "successful_status_code": 200,
        },
        Object {
          "examples": Object {
            "request": Object {
              "address": "Chelsea Market, 75 9th Avenue, New York 10011",
              "avg_cost": "$25",
              "city": "New York City",
              "name": "Los Tacos No. 1",
              "rating": 4,
            },
          },
          "method": "PUT",
          "path": "/api/v1/cities/:id",
          "properties": Object {
            "capital": "boolean",
            "created_at": "timestamp",
            "id": "number",
            "name": "string",
            "population": "number",
            "state": "string",
            "updated_at": "timestamp",
          },
          "request_should_include": Array [
            "name",
            "address",
            "city",
            "rating",
            "avg_cost",
          ],
          "returns": "confirmation message",
          "successful_status_code": 202,
        },
        Object {
          "examples": Object {},
          "method": "DELETE",
          "path": "/api/v1/cities/:id",
          "properties": Object {
            "capital": "boolean",
            "created_at": "timestamp",
            "id": "number",
            "name": "string",
            "population": "number",
            "state": "string",
            "updated_at": "timestamp",
          },
          "request_should_include": "none",
          "returns": "status code only",
          "successful_status_code": 204,
        },
        Object {
          "examples": Object {
            "response": Array [
              Object {
                "address": "2817 East 3rd Avenue, Denver 80206",
                "avg_cost": "$25",
                "city": "Denver",
                "city_id": 19,
                "created_at": "2019-02-05T00:54:21.037Z",
                "id": 204,
                "name": "Machete Tequila + Tacos",
                "rating": 3,
                "updated_at": "2019-02-05T00:54:21.037Z",
              },
            ],
          },
          "method": "GET",
          "path": "/api/v1/restaurants/:id",
          "properties": Object {
            "address": "string",
            "avg_cost": "string",
            "city_id": "number",
            "created_at": "timestamp",
            "id": "number",
            "name": "string",
            "rating": "number",
            "updated_at": "timestamp",
          },
          "request_should_include": "none",
          "returns": "an array of any restaurants in the database that match the id",
          "successful_status_code": 200,
        },
        Object {
          "examples": Object {
            "request": Object {
              "address": "2817 East 3rd Avenue, Denver 80206",
              "avg_cost": "$25",
              "city": "Denver",
              "name": "Machete Tequila + Tacos",
              "rating": 3,
            },
          },
          "method": "PUT",
          "path": "/api/v1/restaurants/:id",
          "properties": Object {
            "address": "string",
            "avg_cost": "string",
            "city_id": "number",
            "created_at": "timestamp",
            "id": "number",
            "name": "string",
            "rating": "number",
            "updated_at": "timestamp",
          },
          "request_should_include": Array [
            "name",
            "address",
            "city",
            "rating",
            "avg_cost",
          ],
          "returns": "all properties with new id",
          "successful_status_code": 201,
        },
        Object {
          "examples": Object {},
          "method": "DELETE",
          "path": "/api/v1/restaurants/:id",
          "properties": Object {
            "address": "string",
            "avg_cost": "string",
            "city_id": "number",
            "created_at": "timestamp",
            "id": "number",
            "name": "string",
            "rating": "number",
            "updated_at": "timestamp",
          },
          "request_should_include": "none",
          "returns": "response code only",
          "successful_status_code": 204,
        },
        Object {
          "examples": Object {
            "response": Array [
              Object {
                "address": "200 Fillmore Street, Denver 80206",
                "avg_cost": "$10",
                "city": "Denver",
                "city_id": 19,
                "created_at": "2019-02-05T00:54:21.038Z",
                "id": 205,
                "name": "Margs Taco Bistro",
                "rating": 3,
                "updated_at": "2019-02-05T00:54:21.038Z",
              },
              Object {
                "address": "9675 Washington Street 80229",
                "avg_cost": "$25",
                "city": "Denver",
                "city_id": 19,
                "created_at": "2019-02-05T00:54:21.038Z",
                "id": 206,
                "name": "Taco John's",
                "rating": 0,
                "updated_at": "2019-02-05T00:54:21.038Z",
              },
              Object {
                "address": "1085 North Broadway 80203",
                "avg_cost": "$25",
                "city": "Denver",
                "city_id": 19,
                "created_at": "2019-02-05T00:54:21.039Z",
                "id": 207,
                "name": "Torchy's Tacos",
                "rating": 3,
                "updated_at": "2019-02-05T00:54:21.039Z",
              },
            ],
          },
          "method": "GET",
          "path": "/api/v1/cities/:id/restaurants",
          "properties": Object {
            "address": "string",
            "avg_cost": "string",
            "city_id": "number",
            "created_at": "timestamp",
            "id": "number",
            "name": "string",
            "rating": "number",
            "updated_at": "timestamp",
          },
          "request_should_include": "none",
          "returns": "an array of restaurants that have matching city ids",
          "successful_status_code": 200,
        },
        Object {
          "examples": Object {
            "request": Object {
              "address": "1085 North Broadway 80203",
              "avg_cost": "$25",
              "city": "Denver",
              "name": "Torchy's Tacos",
              "rating": 3,
            },
          },
          "method": "POST",
          "path": "/api/v1/cities/:id/restaurants",
          "properties": Object {
            "address": "string",
            "avg_cost": "string",
            "city_id": "number",
            "created_at": "timestamp",
            "id": "number",
            "name": "string",
            "rating": "number",
            "updated_at": "timestamp",
          },
          "request_should_include": Array [
            "name",
            "address",
            "rating",
            "avg_cost",
          ],
          "returns": "all properties with id",
          "successful_status_code": 201,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button />,
        <span
          className="endpoint-text-container"
        >
          <p
            className="url-path"
          />
        </span>,
      ],
      "className": "endpoint-item-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="url-path"
          />,
          "className": "endpoint-text-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "url-path",
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button />,
          <span
            className="endpoint-text-container"
          >
            <p
              className="url-path"
            />
          </span>,
        ],
        "className": "endpoint-item-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="url-path"
            />,
            "className": "endpoint-text-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "url-path",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
